third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv5.cpp

```diff
@@ -502,6 +501,11 @@ struct TCGen5MMAScaledOpConversion
               typeConverter->convertType(aTensorTy.getElementType()), rewriter)
               .getBase();
     }
+    mxfpKind mxfpInstKind = getMXFPKind(
+        op.getAType(), op.getBType(), op.getAScale().getType().getElementType(),
+        op.getBScale().getType().getElementType(), transA || transB);
+    bool opKindIsMXFP4 = mxfpInstKind != mxfpKind::mxf8f6f4;
```

Comment:
A later warning could go here instead but it is less obviously related to the K-packing at this point, and a bit more difficult to decipher if the user is unaware of the meaning of transA/B wrt default expectations for fp4 (A x B.T, K-major).